clippy_lints/src/methods/mod.rs

```diff
@@ -4311,6 +4312,31 @@ declare_clippy_lint! {
     "using `Iterator::last` on a `DoubleEndedIterator`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Checks for `NonZero*::new_unchecked(<literal>)` being used in a `const` context.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Using `NonZero*::new_unchecked()` is an `unsafe` function and requires an `unsafe` context. When used with an
+    /// integer literal in a `const` context, `NonZero*::new().unwrap()` will provide the same result with identical
+    /// runtime performances while not requiring `unsafe`.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// const PLAYERS: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(3) };
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// const PLAYERS: NonZeroUsize = NonZeroUsize::new(3).unwrap();
```

Comment:
CI is failing.
```suggestion
    /// ```no_run
    /// use std::num::NonZeroUsize;
    /// const PLAYERS: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(3) };
    /// ```
    /// Use instead:
    /// ```no_run
    /// use std::num::NonZeroUsize;
    /// const PLAYERS: NonZeroUsize = NonZeroUsize::new(3).unwrap();
```